core/internal/api/logging.go

```diff
@@ -39,11 +39,12 @@ func withRetryLogging(
 	return func(ctx context.Context, resp *http.Response, err error) (bool, error) {
 		willRetry, err := policy(ctx, resp, err)
 
-		if willRetry && resp.StatusCode >= 400 {
-			logger.Debug(
-				"Retrying HTTP error",
-				"status", resp.StatusCode,
-			)
+		if willRetry {
+			if resp == nil {
+				logger.Debug("Retrying HTTP error", "status", 0)
```

Comment:
should log reflect that we didnt get a response?

Reply:
Yeah, I wasn't sure what to put there. Suggestions?

Reply:
`"Retrying HTTP error, empty response", "response", resp` ? something like that, which tells us that we hit an empty response when we did this http request, what do you think?

Reply:
Given that we know `resp == nil` I don't know that logging it is useful. But yeah, we should log 'no response' for sure.